src/sentry/search/events/datasets/spans_metrics.py

```diff
@@ -732,6 +734,54 @@ def function_converter(self) -> Mapping[str, fields.MetricsFunction]:
                     ),
                     default_result_type="duration",
                 ),
+                fields.MetricsFunction(
+                    "http_error_count",
+                    snql_metric_layer=lambda args, alias: Function(
+                        "http_error_count",
+                        [
+                            self.resolve_mri(
+                                "span.self_time"
+                            ),  # gets stripped by the mqb transformer
+                            resolve(
+                                UseCaseID.SPANS,
```

Comment:
The best way to do this would be to actually dispatch to the metrics layer the functions without resolution and enable support for `UseCaseID.SPANS`.

Reply:
In addition, this piece of code should be agnostic from the `UseCaseId`, since that is propagated downstream via the `get_series` call.

Reply:
> dispatch to the metrics layer the functions without resolution and enable support for UseCaseID.SPANS.

How do I do this?

I tried to create a DerivedMetric, but since the spans metrics layer actually uses different metrics depending on the query, it would have required creating a derived metric for every combination. So I created the DerivedOp that could take the metric as an argument, but then I needed access to the org_id to resolve the tag key. Passing the org id in directly doesn't work because the QueryBuilder tries to indexer-resolve the strings in the query before the derived op gets expanded. So I ended up moving the resolve calls up to the dataset.